Livewire Best Practices (ReadMe Update)

michael-rubel opened the pull request:
Hey @calebporzio

Continuing the [discussion](https://github.com/livewire/livewire/discussions/4912) I'm adding Livewire Best Practices to ReadMe so it will be useful for those who work with Livewire on a daily basis. Let me know if you want to add/correct something in the list.

P.S. I know the best practice about Form Requests will be [outdated soon](https://github.com/livewire/livewire/commit/27e8de90dfd10dcd61e9e34e9ab486c35488a528), I'll update it on the release of Form Request support in Livewire internally. 😊

michael-rubel committed:
Add Livewire Best Practices to ReadMe

calebporzio commented:
Hey @michael-rubel, that's a great resource!

I'm open to adding it to the README, but would you mind putting it under a heading of "From the community" and you can add the Awesome list in there as well.

Thanks!

(Also, I'm probably going back on the Form Requests addition - there are lots of other gotchas to consider that may not be worth it for users)

michael-rubel committed:
add "from the community" heading

michael-rubel committed:
add "contributors" section

michael-rubel committed:
remove broken badge "all contributors"

michael-rubel committed:
add livewire logo instead

michael-rubel committed:
add dividers for better readability

michael-rubel committed:
swap sections

michael-rubel committed:
add description

michael-rubel committed:
add badges

michael-rubel commented:
Hi @calebporzio!
Redesigned the ReadMe a bit.

- [x] Added the "From the community" heading
- [x] Added logo, description and badges
- [x] Swapped sections by importance
- [x] Removed old "all contributors" badge
- [x] Added an individual section for contributors
- [x] Added dividers for better visual 

It looks much nicer for me now. 😊
Let me know if you want to fix/revert something.

michael-rubel committed:
remove redundant dividers

michael-rubel committed:
convert heres to markdown links

joshhanley commented:
@michael-rubel Caleb is currently off on paternity leave, so might be a little while before he is back and reviewing PRs 🙂
